nanovdb/nanovdb/util/NodeManager.h

```diff
@@ -8,393 +8,311 @@
+    uint8_t* data() { return mBuffer.data(); }
 
-    /// @brief Return a pointer to the root, or NULL if it is uninitialized
-    RootT* root() { return mRoot; }
+    /// @brief Returns a const pointer to the data.
+    ///
+    /// @warning Note that the return pointer can be NULL if the NodeManagerHandle was not initialized
+    const uint8_t* data() const { return mBuffer.data(); }
 
-    /// @brief Return the number of tree nodes at the specified level
-    uint64_t nodeCount(int level) const { return mNodeCount[level]; }
+    /// @brief Returns the size in bytes of the raw memory buffer managed by this NodeManagerHandle's allocator.
+    uint64_t size() const { return mBuffer.size(); }
 
-    /// @brief Return the i'th leaf node
+    /// @brief Returns a const pointer to the NodeManager encoded in this NodeManagerHandle.
     ///
-    /// @warning Never call this method is the NodeManager is un-initialized
-    Node0* leaf( uint32_t i) const { return mLeafs[i]; }
+    /// @warning Note that the return pointer can be NULL if the template parameter does not match the specified grid!
+    template<typename BuildT>
+    const NodeManager<BuildT>* mgr() const { return this->template getMgr<BuildT>(); }
 
-    /// @brief Return the i'th lower internal node
+    /// @brief Returns a pointer to the NodeManager encoded in this NodeManagerHandle.
     ///
-    /// @warning Never call this method is the NodeManager is un-initialized
-    Node1* lower(uint32_t i) const { return mLower[i]; }
+    /// @warning Note that the return pointer can be NULL if the template parameter does not match the specified grid!
+    template<typename BuildT>
+    NodeManager<BuildT>* mgr() { return no_const(this->template getMgr<BuildT>()); }
 
-    /// @brief Return the i'th upper internal node
+    /// @brief Return a const pointer to the NodeManager encoded in this NodeManagerHandle on the device, e.g. GPU
     ///
-    /// @warning Never call this method is the NodeManager is un-initialized
-    Node2* upper(uint32_t i) const { return mUpper[i]; }
-
-private:
-
-    void clear();
-
-    GridT*   mGrid;
-    TreeT*   mTree;
-    RootT*   mRoot;
-    uint64_t mNodeCount[3];
-    Node2**  mUpper;
-    Node1**  mLower;
-    Node0**  mLeafs;
-}; // NodeManager<GridT> class
-
-template<typename GridT>
-NodeManager<GridT>::NodeManager()
-    : mGrid(nullptr)
-    , mTree(nullptr)
-    , mRoot(nullptr)
-    , mNodeCount{0,0,0}
-    , mUpper(nullptr)
-    , mLower(nullptr)
-    , mLeafs(nullptr)
-{
-}
+    /// @warning Note that the return pointer can be NULL if the template parameter does not match the specified grid!
+    template<typename BuildT, typename U = BufferT>
+    typename std::enable_if<BufferTraits<U>::hasDeviceDual, const NodeManager<BuildT>*>::type
+    deviceMgr() const { return this->template getDeviceMgr<BuildT>(); }
 
-template<typename GridT>
-NodeManager<GridT>::NodeManager(GridT &grid)
-    : mGrid(&grid)
```

Comment:
I see that we can no longer create a node manager this way: 
`nanovdb::NodeManager<nanovdb::FloatGrid> nodeMgr(*host_grid);`
Should we make a note in createNodeManager that this is the preferred way to construct a NodeManager now?
```
    auto node_mgr_handle = nanovdb::createNodeManager(*host_grid);
    auto *mgr = node_mgr_handle.mgr<float>();
```